com.unity.ml-agents/Runtime/Academy.cs

```diff
@@ -631,5 +658,13 @@ public void Dispose()
             // Reset the Lazy instance
             s_Lazy = new Lazy<Academy>(() => new Academy());
         }
+
+        /// <summary>
+        /// Check if the input AcademyFixedUpdateStepper belongs to this Academy.
+        /// </summary>
+        internal bool IsStepperOwner(AcademyFixedUpdateStepper stepper)
+        {
+            return stepper.gameObject.Equals(Academy.Instance.m_StepperObject);
```

Comment:
Sounds like you should either user ReferenceEquals, or compare GetInstanceID()
https://answers.unity.com/questions/752382/how-to-compare-if-two-gameobjects-are-the-same-1.html